ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSerdeFactory.java

```diff
@@ -67,20 +81,22 @@ <T> Serde<T> createSerde(
       final boolean isKey) {
     validate(schema);
 
-    final Optional<Schema> physicalSchema = properties.getSchemaId().isPresent() ? Optional.of(
-        SerdeUtils.getAndTranslateSchema(srFactory, properties.getSchemaId()
-            .get(), new ProtobufSchemaTranslator(properties))) : Optional.empty();
+    final Optional<Schema> physicalSchema = getSRSchema(srFactory);
+
+    final ConnectSchema compatibleSchema = getSRSchema(srFactory)
+        .map(s -> (ConnectSchema) ProtobufSchemas.schemaCompatibleWithSR(schema, s))
```

Comment:
With this change, why do we still need https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java#L519-L526?

unrelated: I don't know why github doesn't allow commenting on lines which is not changed in PR...

Reply:
We don't need it anymore. But I am getting other code refactors I'd like to keep away from this so I can fix the regression.